scapy/layers/netbios.py

```diff
@@ -193,9 +193,10 @@ def mysummary(self):
         )
 
     def answers(self, other):
+        trimmed_name = self.RR_NAME[:15]
```

Comment:
> This PR will trim the original hostname being queried when matching requests in answers.

This isn't really doing what you're saying is done. Where you're trimming the answered host name.. I'm unsure how that hostname could be bigger than 15?

Reply:
We are not getting netbios response matches right now with `sr1` unless we trim the hostname to 15 characters prior to sending the netbios request.

I went to see what you had changed in your PR where you implemented the query responses, and I assumed that `self.RR_NAME` is where the query hostname is being saved and that `answers` is what matches requests to responses. I'm probably making a wrong assumption here because I don't really understand how `sr1` finds matches yet.

I'll open it in the debugger tomorrow and gather some screenshots and fix any mistakes I find. Sorry for the confusion.

Reply:
I think I may have mixed up which hostname is in which variable. If `other.QUESTION_NAME` is the original request, then that might be the one that needs trimmed.

Reply:
I think that the best, and most Scapish-way of achieving this behavior would be to add an `any2i()` function to the Netbios field that crops the input to the first 15 characters.
This way when building the packet, you'd already see the name being cropped.

Reply:
Sounds good, I'll look into this and update the PR. Thanks for the input!

Reply:
I think I need another nudge. I have added an `any2i` method to the `NetBIOSNameField` with a couple of print statements to make sure it is running as expected:

```python
class NetBIOSNameField(StrFixedLenField):
    def __init__(self, name, default, length=31):
        # type: (str, bytes, int) -> None
        StrFixedLenField.__init__(self, name, default, length)

    def any2i(self, pkt, x):
        print(x)
        if len(x) > 15:
            x = x[:15]
        print(x)
        return x
```

I can see that it is running the code and trimming the field value correctly:

```
Loremipsumdolorsitamet
Loremipsumdolor
```

However, the response packet is still not being captured.

I see [how it's called in the `Packet.__init__()` method](https://github.com/secdev/scapy/blob/a1afb9a42704873767015246522b10ed86ce35b1/scapy/packet.py#L197C13-L197C39), and I'm trying to look at other examples in `fields.py` but I'm a little lost at the moment.

Reply:
I've also tried fiddling with the `length=31` parameter to change the length of the `StrFixedLenField` to 15, but that resulted in a malformed packet.

Reply:
Oh sorry, `h2i` might be more appropriate

Reply:
`h2i` was what we needed. Thanks for the help! The most recent commit was working for me when testing.

Reply:
That's great !